### PR TITLE
Issue 623 Cloud Pipeline shall allow to "block" the user

### PR DESCRIPTION
**Pull request by Wedds**
PR solves issue #623
It allows blocking a user to deny his access to the account. 

- `blocked` field is added to PipelineUser model and to user table in DB as well as flyway migration script
- Handling of `blocked` field is implemented in UserDao and related XML-file
- `updateUserBlockingStatus` method added to API in UserController
- Handling of `blocked` user status for authorization step is provided at SAMLUserDetailsServiceImpl class
- Necessary tests for new functionality of UserManager and SAMLUserDetailsServiceImpl classes were specified

### CODE REVIEW

File: api/src/main/java/com/epam/pipeline/controller/user/UserController.java [hunk 1]
```diff
@@ -222,6 +222,19 @@ public Result deleteUser(@PathVariable Long id) {
         return Result.success(userApiService.getUserControls());
     }
 
+    @RequestMapping(value = "/user/{id}/block", method = RequestMethod.PUT)
```
Comment (ekazachkova) [inline api/src/main/java/com/epam/pipeline/controller/user/UserController.java]: Could we use `PutMapping` annotation here?

File: api/src/main/java/com/epam/pipeline/manager/user/UserManager.java [hunk 2]
```diff
@@ -157,6 +157,13 @@ public PipelineUser updateUser(Long id, PipelineUserVO userVO) {
         return userDao.updateUser(user);
     }
 
+    @Transactional(propagation = Propagation.REQUIRED)
+    public PipelineUser updateUserBlockingStatus(Long id, final boolean blockStatus) {
```
Comment (ekazachkova) [inline api/src/main/java/com/epam/pipeline/manager/user/UserManager.java]: Please, make final method arguments and local fields

File: api/src/main/resources/db/migration/v2019.08.24_12.00__Issue623_add_user_blocking_support.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE pipeline.user ADD blocked boolean DEFAULT false;
```
Comment (ekazachkova) [inline api/src/main/resources/db/migration/v2019.08.24_12.00__Issue623_add_user_blocking_support.sql]: Please, don't forget to add empty line at the end of the file

File: api/src/test/java/com/epam/pipeline/manager/user/UserManagerTest.java [hunk 4]
```diff
@@ -64,4 +110,23 @@ public void deleteUser() {
         userManager.deleteUser(user.getId());
         Assert.assertTrue(notificationDao.loadAllNotifications().isEmpty());
     }
+
+    private void compareAllFieldOfUsers(PipelineUser firstUser, PipelineUser secondUser) {
```
Comment (ekazachkova) [inline api/src/test/java/com/epam/pipeline/manager/user/UserManagerTest.java]: The method arguments names are a little bit unclear. I would prefer to rename args:
first -> expected
second -> actual
Also, could you make method arguments final.

**Review by ekazachkova — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/epam/pipeline/controller/user/UserController.java (hunk 1 above)
Comment (Wedds) [reply api/src/main/java/com/epam/pipeline/controller/user/UserController.java]: Done

File: api/src/main/java/com/epam/pipeline/manager/user/UserManager.java (hunk 2 above)
Comment (Wedds) [reply api/src/main/java/com/epam/pipeline/manager/user/UserManager.java]: Done

File: api/src/main/resources/db/migration/v2019.08.24_12.00__Issue623_add_user_blocking_support.sql (hunk 3 above)
Comment (Wedds) [reply api/src/main/resources/db/migration/v2019.08.24_12.00__Issue623_add_user_blocking_support.sql]: Done

File: api/src/test/java/com/epam/pipeline/manager/user/UserManagerTest.java (hunk 4 above)
Comment (Wedds) [reply api/src/test/java/com/epam/pipeline/manager/user/UserManagerTest.java]: Done

**Review by ekazachkova — APPROVED** [review]

**Review by mzueva — CHANGES_REQUESTED** [review]
Test `RoleDaoTest.testLoadRolesWithUsers()` fails, please check RoleDao queries.

**Review by mzueva — CHANGES_REQUESTED** [review]
The same changes are required for `loadRoleWithUsersQuery`

**Review by mzueva — APPROVED** [review]

**Pull request merged**